internal/adapters/supabase-purchase-repository.adapter.go

```diff
@@ -54,3 +69,23 @@ func (r *SupabasePurchaseRepository) FindById(id string, where myTypes.Where) (*
 	panic("implement me")
 }
 
+func (r *SupabasePurchaseRepository) FindAll(where myTypes.Where) ([]models.PurchaseModel, error) {
+	var supabaseData []myTypes.AnyMap
+
+	query := r.supabaseClient.DB.From("purchases").Select("*")
```

Comment:
Para selecionar também os itens, é algo mais ou menos assim:
```suggestion
	query := r.supabaseClient.DB.From("purchases").Select("*", "items (*)")
```